src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -472,6 +484,13 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
     std::vector<QueueSubmission> completed_submissions =
         PullCompletedSubmissions(device, query_pool);
 
+    // Completing a submission is not possible when a timestamp is not yet ready to be queried (and
+    // this is in the hands of Vulkan). We therefore maintain a list of submissions to be retried
+    // later and append this list here for the current round.
+    completed_submissions.insert(completed_submissions.end(), retry_to_complete_.begin(),
```

Comment:
Is there are reason to have `retry_to_complete_` as a member and add them to the completed submissions at the beginning of the method here?
I'd guess it would be a bit more straightforward to have a local vector here, and add it to the `queue_to_submissions` at the end of the method?

Reply:
Yes. The issue is that queue_to_submissions_ is a map from queue to submissions and at this point I don't know the queue anymore. We could plumb this through in PullCompletedSubmissions, etc., but using this additional vector seemed to be the easier solution.

Reply:
Could we not just add the queue to the `QueueSubmission` struct? I guess that would be cleaner.

Reply:
Done.